src/apps/bastion-protocol/helper/bastion-protocol.supply.token-helper.ts

```diff
@@ -0,0 +1,148 @@
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+  ) { }
+
+  async getTokens<T = BastionProtocolComptroller, V = BastionProtocolCtoken>({
+    comptrollerAddress,
+    realmName,
+    network,
+    appId,
+    groupId,
+  }: BastionSupplyTokenHelperParams<T, V>) {
+    const multicall = this.appToolkit.getMulticall(network);
+    const comptrollerContract =
+      this.bastionProtocolContractFactory.bastionProtocolComptroller(
+        {
+          address: comptrollerAddress,
+          network
+        }
+      );
+    const marketTokenAddressesRaw = await multicall.wrap(comptrollerContract).getAllMarkets();
+
+    const tokens = await Promise.all(
+      marketTokenAddressesRaw.map(async marketTokenAddressRaw => {
+        const address = marketTokenAddressRaw.toLowerCase();
+        const erc20TokenContract = this.bastionProtocolContractFactory.erc20({ address, network });
+        const contract = this.bastionProtocolContractFactory.bastionProtocolCtoken({ address, network });
+
+        const underlyingAddress = await multicall.wrap(contract).underlying()
+          .then(t => t.toLowerCase().replace(ETH_ADDR_ALIAS, ZERO_ADDRESS))
+          .catch(() => ZERO_ADDRESS);
+
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+        const allTokens = [...baseTokens];
+
+        const underlyingToken = allTokens.find(v => v.address === underlyingAddress);
+        if (!underlyingToken) return null;
+
+        const [symbol, decimals, supplyRaw, rateRaw, supplyRateRaw, borrowRateRaw] = await Promise.all([
+          multicall.wrap(erc20TokenContract).symbol(),
+          multicall.wrap(erc20TokenContract).decimals(),
+          multicall.wrap(erc20TokenContract).totalSupply(),
+          multicall.wrap(contract).callStatic.exchangeRateCurrent(),
+          multicall.wrap(contract).supplyRatePerBlock(),
+          multicall.wrap(contract).borrowRatePerBlock(),
+        ]);
+
+        // Data Props
+        const type = ContractType.APP_TOKEN;
+        const supply = Number(supplyRaw) / 10 ** decimals;
+        const underlyingTokenDecimals = underlyingToken.decimals;
+        // The exchange rate has (18 - 8 + UnderlyingTokenDecimals) decimals
+        const mantissa = underlyingTokenDecimals + 10;
+        const pricePerShare = Number(rateRaw) / 10 ** mantissa;
+        const price = pricePerShare * underlyingToken.price;
+        const liquidity = price * supply;
+        const tokens = [underlyingToken];
+        const secondsPerDay = 60 * 60 * 24;
+        const supplyRate = formatUnits(supplyRateRaw, 18);
+        const borrowRate = formatUnits(borrowRateRaw, 18);
+
+        const supplyApy = Math.pow(1 + secondsPerDay * Number(supplyRate), 365) - 1;
+        const borrowApy = Math.pow(1 + secondsPerDay * Number(borrowRate), 365) - 1;
+
+        // Display Props
+        const label = underlyingToken.symbol;
```

Comment:
You should use the `realmName` in the label. The `dataProps` are not displayed in the UI, only the `displayProps`.

Reply:
Example: `const label = `${underlyingToken.symbol} in ${realmName}`

Reply:
Did just that, thanks for letting me know.